src/components/accessibility/partitions_data_table.tsx

```diff
@@ -0,0 +1,165 @@
+import { GlobalChartState } from '../../state/chart_state';
+import {
+  A11ySettings,
+  DEFAULT_A11Y_SETTINGS,
+  getA11ySettingsSelector,
+} from '../../state/selectors/get_accessibility_config';
+import { getInternalIsInitializedSelector, InitStatus } from '../../state/selectors/get_internal_is_intialized';
+import { ValueFormatter } from '../../utils/common';
+
+interface ScreenReaderPartitionTableProps {
+  a11ySettings: A11ySettings;
+  screenReaderData: LabelsInterface[];
+  shapeViewModel: ShapeViewModel[];
+  formatter?: ValueFormatter;
+  configMaxCount?: number;
+}
+
+const renderTableRows = (
+  value: LabelsInterface,
+  index: number,
+  count: number,
+  configMaxCount?: number,
+  formatter?: ValueFormatter,
+) => {
+  return (
+    <tr
+      key={`row--${index}`}
+      id={configMaxCount && configMaxCount * (count - 1) === index ? `startOfConfigMaxCount` : undefined}
+    >
+      <th scope="row">{value.label}</th>
+      <td>{formatter && formatter(value.valueText) ? formatter(value.valueText) : value.valueText}</td>
+      <td>{value.percentage}</td>
+    </tr>
+  );
+};
+
+const showOnlyLimitedRows = (
+  configMaxCount: number,
+  count: number,
+  data: LabelsInterface[],
+  formatter?: ValueFormatter,
+) => {
+  return data.map((value: LabelsInterface, i: number) => {
+    return renderTableRows(value, i, count, configMaxCount, formatter);
+  });
+};
+
+const renderTableContent = (d: any[], count: number, formatter?: ValueFormatter, configMaxCount?: number) => {
+  return d.length < 200 || !configMaxCount
+    ? d.map((value: LabelsInterface, i: number) => {
+        return renderTableRows(value, i, count, configMaxCount, formatter);
+      })
+    : showOnlyLimitedRows(configMaxCount, count, d.slice(0, count * configMaxCount), formatter);
+};
+
+const ScreenReaderPartitionTableComponent = ({
+  a11ySettings,
+  shapeViewModel,
+  screenReaderData,
+  formatter,
+  configMaxCount,
+}: ScreenReaderPartitionTableProps) => {
+  const [count, setCount] = useState(1);
+  const { tableCaption } = a11ySettings;
+
+  const handleMoreData = () => {
+    setCount(count + 1);
+    const nextSliceOfData = screenReaderData.slice(count * configMaxCount!, count * configMaxCount! + configMaxCount!);
+
+    // generate the next group of data
+    return showOnlyLimitedRows(configMaxCount!, count, nextSliceOfData, formatter);
+  };
+
+  const showMoreCellsButton =
+    configMaxCount && screenReaderData.length > 200 ? (
+      <tfoot>
+        <tr>
+          <td>
+            <button type="submit" key={Math.random()} onClick={() => handleMoreData()} tabIndex={-1}>
+              Click to show more data
+            </button>
+          </td>
+        </tr>
+      </tfoot>
+    ) : null;
+
+  return (
+    <div className="echScreenReaderOnly echScreenReaderTable">
+      <table>
+        <caption>
+          {tableCaption === 'Table of the data for screen reader navigation'
```

Comment:
looks like this string `'Table of the data for screen reader navigation'` is not even used for the caption.
So I think we can make the `tableCaption` optional and align the ternary operator accordingly.
This code can lead to bugs if the default value in `src/specs/constants.ts` is changed without updating this string

Reply:
👍🏻  71050c0 for changes!

Reply:
We should check if `tableCaption` is undefined (if you want double check for null) using `isNil` if for whatever reason the user wants to hide the `tableCaption` using a simple empty string
```suggestion
{isNil(tableCaption) ? "default text" : tableCaption}
```

Reply:
👍🏻  e654f11 for changes!